### PR TITLE
Implement 2D version of masked_select for nestedtensors

### PR DESCRIPTION
**Pull request by krzysztofjordan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133889

### CODE REVIEW

**Comment by krzysztofjordan** [comment]
Accidentally did the wrong process for https://github.com/pytorch/pytorch/pull/131069, so putting this up as a landable diffs

**Review by soulitzer — APPROVED** [review]

File: torch/nested/__init__.py [hunk 1]
```diff
@@ -410,3 +411,55 @@ def nested_tensor_from_jagged(
...
+    >>> mask = torch.tensor([False])
+    >>> nt = torch.nested.masked_select(tensor, mask)
+    >>> nt.shape
+    torch.Size([6, j5])
+    >>> # Length of each item in the batch:
+    >>> nt.offsets().diff()
+    tensor([0, 0, 0, 0, 0, 0])
+    """
+    if tensor.layout != torch.strided:
+        raise RuntimeError(
```
[the remark this answers is not in the document]
Comment (soulitzer) [reply torch/nested/__init__.py:446]: This is a good point, though we already have RuntimeError for everything else in this file, and technically its somewhat bc-breaking to change everything to now raise ValueError.

[the remark this answers is not in the document]
Comment (krzysztofjordan) [reply torch/nested/__init__.py:446]: We could file a BE task to update this

**Pull request closed without merging**